File: 백준/Silver/2890. 카약/카약.c
#include <stdio.h>

int main(void) { 
   int height, length, flag, count = 1;
   scanf("%d %d", &height, &length);

   char map[height][length];
   for(int i = 0; i < height; i++){
      scanf("%s", map[i]);
   }
   
   char rank[10] = "000000000\0";
   for(int i = length - 2; i > 0; i--){
      flag = 0;
      for(int j = 0; j < height; j++){
         char c = map[j][i];
         if(c != '.' && rank[c - '0' - 1] == '0'){
            int k = c - '0';
            rank[k - 1] = count + '0';
            flag = 1;
         }
      }
      if(flag) count++;
   }

   for(int i = 0; i < 9; i++){
      printf("%c\n", rank[i]);
   }
   
   
   return 0;
}

File: 백준/Silver/2890. 카약/카약.py
height, length = map(int, input().split())

maps = []

for _ in range(height):
  maps.append(input())

rank = [0 for _ in range(9)]

count = 1
for i in range(length - 2, 0, -1):
  flag = False
  for j in range(height):
    if maps[j][i] != '.' and rank[int(maps[j][i]) - 1] == 0:
      rank[int(maps[j][i]) - 1] = count
      flag = True
  if flag:
    count += 1

for item in rank:
  print(item)